src/proxy-ui-api/frontend/src/store/modules/notifications.ts

```diff
@@ -28,41 +28,104 @@ import { RootState } from '../types';
 import { Notification } from '@/ui-types';
 
 export interface NotificationsState {
-  successMessageCode: string;
-  successMessageRaw: string;
-  showSuccessCode: boolean;
-  showSuccessRaw: boolean;
-  notifications: Notification[];
+  errorNotifications: Notification[];
+  successNotifications: Notification[];
 }
 
 const getDefaultState = () => {
   return {
-    successMessageCode: '',
-    successMessageRaw: '',
-    showSuccessCode: false,
-    showSuccessRaw: false,
-    notifications: [],
+    errorNotifications: [],
+    successNotifications: [],
   };
 };
 
+// Finds if an array of notifications contains a similar notification.
+function containsNotification(
+  errorNotifications: Notification[],
+  notification: Notification,
+): number {
+  if (!notification || !errorNotifications || errorNotifications.length === 0) {
```

Comment:
I think you could use the custom isEmpty -utility method from util/helpers.ts here.

So the clause could be written like this (double check!):

if (isEmpty(notification) || isEmpty(errorNotifications)) {

Reply:
isEmpty accepts strings or arrays. And seems that it's currently used only for strings. I'm not sure if it would be good to expand it for Notifications or some more general things.

Reply:
Ah yes, I've that notification is an object. But I guess it could be used for errorNotifications

Reply:
I suppose the line would look like this  `if (!notification || isEmpty(errorNotifications as [])) {`  
and import of the isEmpty would add one line. So it seems a bit boilerplated and I would rather keep it like it is now.

Related to the isEmpty method, I think there is a case to make it to accept string only, or somehow define the array type smarter.